### DISCUSSION TITLE
Why are parameters duplicated in action documents?

### DISCUSSION THREAD

**Discussion by MatthewStrickland** [StackStorm/st2#5954] [General]
I noticed the documents inside action_execution_d_b have the json paths of **parameters** & **liveaction.parameters**. In every instance I've checked they hold the same data (the input params to the action). This at least is causing an issue for me because my parameter string is quite large and it's being duplicated, doubling the storage needed and hitting the 16MB Document limit in mongo.

What is the reasoning for this and what is liveaction all about? Is there any way for me to avoid storing it twice in the same object?

**Comment by guzzijones** [comment]
I am working on removing this duplicate embedded document. [PR](https://github.com/StackStorm/st2/pull/5995).  I have all the unit tests passing.

**Comment by guzzijones** [comment]
each attribute is most likely a document so it shouldn't affect the 16mb limit.
